packages/builder/src/builtin-tasks/migrate.ts

```diff
@@ -0,0 +1,59 @@
+import debug from "debug";
+import fsExtra from "fs-extra";
+import path from "path";
+import fs from "fs";
+
+import { task } from "../internal/core/config/config-env";
+import { BuilderError } from "../internal/core/errors";
+import { ERRORS } from "../internal/core/errors-list";
+import { runScriptWithAlgob } from "../internal/util/scripts-runner";
+import { TASK_MIGRATE } from "./task-names";
+import { glob } from "./util";
+import { RuntimeArgs } from "../types";
+import { runSingleScript } from "./run";
+
+export default function () : void {
+  const log = debug("builder:core:tasks:migrate");
+
+  task(TASK_MIGRATE, "Compiles and runs user-defined scripts from scripts directory")
+    .addPositionalParam(
+      "directory",
+      "A directory that contains js files to be run within builder's environment",
+      "scripts"
+    )
+    .addFlag("noCompile", "Don't compile before running this task")
+    .setAction(
+      async (
+        { directory, noCompile }: { directory: string; noCompile: boolean },
+        { run, runtimeArgs }
+      ) => {
+
+        if (!fs.existsSync(directory)) {
+          throw new BuilderError(ERRORS.BUILTIN_TASKS.MIGRATE_DIRECTORY_NOT_FOUND, {
+            directory,
+          });
+        }
+
+        const scriptNames = (await glob(path.join(directory, "*.js"), {})).sort()
```

Comment:
if you move this outside, then you can test it.

Reply:
But there is no way to test it because the filesystem always sorts them in my case. So either I have to omit the `glob` part because it loads filenames directly or I have to only pass filenames to sorter which is known to work fine.

Reply:
I don't understand the claim about the filesystem. It doesn't matter if filesystem sorts it or not - because you call a JS sort function at the end. 

So the test is simple - to check if indeed files are sorted with JS

Reply:
Sorting order is already checked in the other test because the `output.txt` is populated in a particular way and it's contents are checked.
I added a unit test to check this particular logic. I don't think it was needed because the output test covers this but I added it.

Reply:
Moved to it's own method in the `builtin-tasks/util.ts` file.